src/api/c/filters.cpp

```diff
@@ -19,17 +20,68 @@
 using af::dim4;
 using namespace detail;
 
+af_err af_medfilt(af_array *out, const af_array in, const dim_t wind_length, const dim_t wind_width, const af_border_type edge_pad)
+{
+    return af_medfilt2(out, in, wind_length, wind_width, edge_pad);
```

Comment:
Can you check the dimensions of `in` like @shehzan10 mentioned and call `af_medfilt1` if it is 1D ?